sklearn/neighbors/tests/test_lof.py

```diff
@@ -44,14 +47,15 @@ def test_lof():
 
     # Assert predict() works:
     clf = neighbors.LocalOutlierFactor(contamination=0.25, n_neighbors=5).fit(X)
-    assert_array_equal(clf._predict(), 6 * [1] + 2 * [-1])
-    assert_array_equal(clf.fit_predict(X), 6 * [1] + 2 * [-1])
+    expected_predictions = np.asarray(6 * [1] + 2 * [-1], dtype=global_dtype)
+    assert_array_equal(clf._predict(), expected_predictions)
+    assert_array_equal(clf.fit_predict(X), expected_predictions)
 
 
-def test_lof_performance():
+def test_lof_performance(global_dtype):
     # Generate train/test data
     rng = check_random_state(2)
-    X = 0.3 * rng.randn(120, 2)
+    X = 0.3 * rng.randn(120, 2).astype(global_dtype, copy=False)
     X_train = X[:100]
 
     # Generate some abnormal novel observations
```

Comment:
X_outliers should be cast as well otherwise it will always be float64 and thus X_test also